<?xml version="1.0"?>
<robot name="my_robot">

    <material name="pink">
        <color rgba="1.0 0.8 0.8 1.0"/>
    </material>
    <material name="grey">
        <color rgba="0.8 0.8 0.8 1.0"/>
    </material>

    <link name="base">
        <visual>
            <geometry>
                <box size="5 5 0.01" />
            </geometry>
            <material name="pink"/>
        </visual>
    </link>

    <link name="back">
        <visual>
            <geometry>
                <box size="10 5 0.01" />
            </geometry>
            <origin rpy="0 1.57075 0" xyz="0 0 5"/>
            <material name="pink"/>
        </visual>
    </link>

    <joint name="base_to_back" type="fixed">
        <parent link="base"/>
        <child link="back"/>
        <origin xyz="-2.5 0 0"/>
    </joint>

    <link name="right">
        <visual>
            <geometry>
                <box size="5 10 0.01" />
            </geometry>
            <origin rpy="1.57075 0 0" xyz="0 0 5" />
            <material name="pink"/>
        </visual>
    </link>

    <joint name="base_to_right" type="fixed">
        <parent link="base"/>
        <child link="right"/>
        <origin xyz="0 2.5 0"/>
    </joint>

    <link name="left">
        <visual>
            <geometry>
                <box size="5 10 0.01" />
            </geometry>
            <origin rpy="1.57075 0 0" xyz="0 0 5" />
            <material name="pink"/>
        </visual>
    </link>

    <joint name="base_to_left" type="fixed">
        <parent link="base"/>
        <child link="left"/>
        <origin xyz="0 -2.5 0"/>
    </joint>

    <link name="upper">
        <visual>
            <geometry>
                <box size="5 5 0.01" />
            </geometry>
            <origin xyz="2.5 0 0" />
            <material name="pink"/>
        </visual>
    </link>

    <joint name="back_to_upper" type="fixed">
        <parent link="back"/>
        <child link="upper"/>
        <origin xyz="0 0 10"/>
    </joint>


    <link name="link_1">
        <visual>
            <geometry>
                <box size="0.5 5 0.5" />
            </geometry>
            <origin xyz="0 2.5 0" />
            <material name="grey"/>
        </visual>
    </link>

    <joint name="left_to_link_1" type="prismatic">
        <parent link="left"/>
        <child link="link_1"/>
        <origin xyz="0 0 5"/>
        <limit effort="1000.0" lower="-1.9" upper="1.9" velocity="0.5" />
        <axis xyz="1 0 0" />
    </joint>

    <link name="hidden_link">
        <visual>
            <geometry>
                <box size="0.5 0.5 0.5" />
            </geometry>
        </visual>
    </link>

    <joint name="link_1_to_hidden_link" type="prismatic">
        <parent link="link_1"/>
        <child link="hidden_link"/>
        <origin xyz="0 2.5 0"/>
        <limit effort="1000.0" lower="-1.9" upper="1.9" velocity="0.5" />
        <axis xyz="0 1 0" />
    </joint>

    <link name="link_2">
        <visual>
            <geometry>
                <box size="0.5 0.5 5" />
            </geometry>
            <origin xyz="0 0 2.5" />
            <material name="grey"/>
        </visual>
    </link>

    <joint name="hidden_link_to_link_2" type="prismatic">
        <parent link="hidden_link"/>
        <child link="link_2"/>
        <origin xyz="0.5 0 0"/>
        <limit effort="1000.0" lower="0" upper="4" velocity="0.5" />
        <axis xyz="0 0 -1" />
    </joint>

    <link name="left_gripper">
        <visual>
            <geometry>
                <mesh filename="package://pr2_description/meshes/gripper_v0/l_finger.dae"
                scale="10 10 10"/>
            </geometry>
            <origin rpy="0 1.57075 3.14159" xyz="0 0 0" />
        </visual>
    </link>

    <joint name="link_2_to_left_gripper" type="revolute">
        <parent link="link_2"/>
        <child link="left_gripper"/>
        <origin xyz="0 0 0"/>
        <limit effort="1000.0" lower="0" upper="1.57075" velocity="0.5" />
        <axis xyz="-1 0 0" />
    </joint>

    <link name="right_gripper">
        <visual>
            <geometry>
                <mesh filename="package://pr2_description/meshes/gripper_v0/l_finger.dae"
                scale="10 10 10"/>
            </geometry>
            <origin rpy="0 1.57075 0" xyz="0 0 0" />
        </visual>
    </link>

    <joint name="link_2_to_right_gripper" type="revolute">
        <parent link="link_2"/>
        <child link="right_gripper"/>
        <origin xyz="0 0 0"/>
        <limit effort="1000.0" lower="0" upper="1.57075" velocity="0.5" />
        <axis xyz="1 0 0" />
    </joint>

</robot>